src/settings-ui/PowerToys.Settings/MainWindow.xaml.cs

```diff
@@ -42,6 +39,20 @@ public static void CloseHiddenWindow()
             }
         }
 
+        public void Open(Type type)
+        {
+            if (inst != null)
+            {
+                this.Activate();
+            }
+            else
+            {
+                this.Show();
```

Comment:
If you call `settingsWindow.Show()` before calling `settingsWindow.Open(type);`, the `inst` will not be null and this `else` will not be necessary.
It's still OK to check for `inst != null` and put `ShellPage.Navigate(type);` inside the `if` as well.